src/cargo.rs

```diff
@@ -194,48 +197,34 @@ pub(crate) fn select_buildopt<'a>(
     opts: impl IntoIterator<Item = &'a BuildOpt>,
     cmd_kind: CmdKind,
 ) -> Result<&'a BuildOpt, Error> {
-    let opts = opts.into_iter();
-
     // Target kinds we want to look for
     let look_for = &[TargetKind::Bin, TargetKind::Example, TargetKind::Test];
-    let is_test = cmd_kind == CmdKind::Test;
 
     // Find candidates with the possible target types
-    let mut candidates = opts
+    let candidates: Vec<_> = opts
+        .into_iter()
         .filter(|opt| {
-            // When run as a test we only care about the binary where the profile
-            // is set as `test`
-            if is_test {
-                opt.profile.test
-            } else {
-                opt.target
+            // When run as a test or bench we only care about the
+            // binary where the profile is set as `test`
+            match cmd_kind {
+                CmdKind::Test | CmdKind::Bench => opt.profile.test,
```

Comment:
I'm guessing you tested and the test property was set to true when running benchmarks?

Reply:
Indeed, turns out having small projects is good for testing (see #11 ). Who would have thought :D